### PR TITLE
Minimal for resnet vizualized

### PR DESCRIPTION
**Pull request by ljfitz**
I didn't make any attempt to get the table properties filed in properly, but did just enough to have most nodes show up.  
Note it didn't work entirely, as it only shows `softmax` and not `argmax`, but the piecewise improvement is worth having:
image

For reference - netron shows:
image

### CODE REVIEW

File: lib/Transform/ATenVisualGraph.cpp [hunk 1]
```diff
@@ -704,6 +704,16 @@ struct ATenVisualGraphPass : public ATenVisualGraphBase<ATenVisualGraphPass> {
       fillPropertiesSliceOp(op2, std::move(props));
     } else if (auto op2 = dyn_cast<Torch::AtenConstantPadNdOp>(op)) {
       props.appendIntList("Attributes.padding", op2.pad());
+    } else if (auto op2 = dyn_cast<Torch::AtenMeanDimOp>(op)) {
+      // props.appendIntList("Attributes.padding", op2.pad());
```
Comment (gargaroff) [inline lib/Transform/ATenVisualGraph.cpp:708]: Why is most of the new code commented out? I usually don't like introducing commented out code, as it tends to get forgotten and bit-rot, but if there's a valid reason & precedence for it, then I'm fine with it.

File: lib/Transform/operators_supported.json [hunk 2]
```diff
@@ -27,6 +27,22 @@
 		["Storage.Bytes", "Total storage required for MaxPool2D engine", "long"]
 	    ]		
 	},
+	{
+	    "name" : "torch.aten.mean.dim",
```
Comment (gargaroff) [inline lib/Transform/operators_supported.json:31]: Unrelated to the review, just to get it out there: I was thinking whether we might be able to generate this information using existing TableGen definitions. Let's discuss this in a separate context if you feel like this might be worthwhile.

**Review by gargaroff — APPROVED** [review]
Just one minor nit

File: lib/Transform/ATenVisualGraph.cpp [hunk 3]
```diff
@@ -704,6 +704,16 @@ struct ATenVisualGraphPass : public ATenVisualGraphBase<ATenVisualGraphPass> {
...
+    } else if (auto op2 = dyn_cast<Torch::AtenSqueezeDimOp>(op)) {
+      // props.appendIntList("Attributes.padding", op2.pad());
+    } else if (auto op2 = dyn_cast<Torch::AtenMmOp>(op)) {
+      // props.appendIntList("Attributes.padding", op2.pad());
+    } else if (auto op2 = dyn_cast<Torch::Aten_SoftmaxOp>(op)) {
+      // props.appendIntList("Attributes.padding", op2.pad());
+    } else if (auto op2 = dyn_cast<Torch::AtenArgmaxOp>(op)) {
+      // props.appendIntList("Attributes.padding", op2.pad());
```
Comment (ljfitz) [inline lib/Transform/ATenVisualGraph.cpp:716]:
```suggestion
    } else if (auto op2 = dyn_cast<Torch::AtenMeanDimOp>(op)) {
      // TODO: fill properties
    } else if (auto op2 = dyn_cast<Torch::AtenSqueezeDimOp>(op)) {
      // TODO: fill properties
    } else if (auto op2 = dyn_cast<Torch::AtenMmOp>(op)) {
      // TODO: fill properties
    } else if (auto op2 = dyn_cast<Torch::Aten_SoftmaxOp>(op)) {
      // TODO: fill properties
    } else if (auto op2 = dyn_cast<Torch::AtenArgmaxOp>(op)) {
      // TODO: fill properties
```

File: lib/Transform/ATenVisualGraph.cpp [hunk 4]
```diff
@@ -714,6 +724,8 @@ struct ATenVisualGraphPass : public ATenVisualGraphBase<ATenVisualGraphPass> {
       fillPropertiesOpC2dActMaxpool(op2, "aten.lrelu", std::move(props));
     } else if (auto op2 = mlir::dyn_cast<xten::AddOp>(op)) {
       // fillPropertiesBinaryALUOp<xten::AddOp>(xtenAddOp, std::move(props));
+    } else if (auto op2 = mlir::dyn_cast<xten::MMOp>(op)) {
+      // fillPropertiesBinaryALUOp<xten::MMOp>(xtenAddOp, std::move(props));
```
Comment (ljfitz) [inline lib/Transform/ATenVisualGraph.cpp:728]:
```suggestion
    } else if (auto op2 = mlir::dyn_cast<xten::MMOp>(op)) {
      // TODO: fill properties
```

File: lib/Transform/operators_supported.json (hunk 2 above)
Comment (ljfitz) [reply lib/Transform/operators_supported.json:31]: I like the idea. Sadly `torch-mlir` may be just about the worst dialect for this. Its operations are generated using `third-party/torch-mlir/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py` which doesn't provide any descriptive information about the operands other than the type. The operation origins are in the `pytorch` op registry which may have more information, but I confess I only know of the connection and don't know the details.

File: lib/Transform/ATenVisualGraph.cpp (hunk 1 above)
Comment (ljfitz) [reply lib/Transform/ATenVisualGraph.cpp:708]: You're right - I had adopted the behavior of the code I copied whereas a `TODO` is clearer in intent. I'll adjust.
